### DISCUSSION TITLE
PanelMenu problem, link reload

### DISCUSSION THREAD

**Discussion by StevanZivadinovic** [primefaces/community#147] [PrimeFaces] [answered]
Hi, when I use PanelMenu component, and add url parameter for each link, now, page reload when i click on different links. I added url parameter because I want when I right-clicked on item, and go on 'open in new browser' to open that link in new tab. So, is there property that prevent reload page when click on unclicked item. I also want to keep url prop, because right-click functionality...Thank you in advance..

**Comment by FlipWarthog** [comment]
On your `p:menuitem` you can add `target="_blank"`, which will tell the browser to open in a new tab if the user left-clicks the item.

```xml
<p:menuitem value="My External Link"
	url="https://google.com" icon="fas fa-up-right-from-square" target="_blank" ajax="false"
	iconPos="right" />
```

**Reply by melloware** [reply]
I could be wrong but I think he is saying he only want the right click "Open In new Tab" to work but it NOT to be a link when they just left click the menu item right?

**Reply by StevanZivadinovic** [reply]
I want to:
1.  right click "Open In new Tab" to work
2. Not to reload page when i click on menu item

**Reply by melloware** [reply]
@StevanZivadinovic can you put together a small PrimeFaces Test reproducer so we can debug.

**Reply by FlipWarthog** [reply, marked as answer]
Since the `menuitem`s render as `a` elements, you might need `onclick` with `preventDefault`, like described here:
https://stackoverflow.com/questions/72870959/how-to-create-anchor-tag-link-that-works-only-on-right-click
